source/standalone/workflows/skrl/play.py

```diff
@@ -0,0 +1,204 @@
+
+    # configure and instantiate PPO agent
+    # https://skrl.readthedocs.io/en/latest/modules/skrl.agents.ppo.html
+    agent_cfg = PPO_DEFAULT_CONFIG.copy()
+    experiment_cfg["agent"]["rewards_shaper"] = None  # avoid 'dictionary changed size during iteration'
+    agent_cfg.update(convert_skrl_cfg(experiment_cfg["agent"]))
+
+    agent_cfg["state_preprocessor_kwargs"].update({"size": env.observation_space, "device": env.device})
+    agent_cfg["value_preprocessor_kwargs"].update({"size": 1, "device": env.device})
+    agent_cfg["experiment"]["checkpoint_interval"] = 0  # don't generate checkpoints
+
+    agent = PPO(
+        models=models,
+        memory=None,  # memory is optional during evaluation
+        cfg=agent_cfg,
+        observation_space=env.observation_space,
+        action_space=env.action_space,
+        device=env.device,
+    )
+
+    # specify directory for logging experiments (load checkpoint)
+    print(f"[INFO] Loading experiment from directory: {log_checkpoint_path}")
+    # get checkpoint path
+    if args_cli.checkpoint:
+        resume_path = os.path.abspath(args_cli.checkpoint)
+    else:
+        resume_path = get_checkpoint_path(log_checkpoint_path, os.path.join("*", "checkpoints"), None)
+    print(f"[INFO] Loading model checkpoint from: {resume_path}")
+    agent.load(resume_path)
+
+    # test the agent according to the selected mode defined with "use_api":
+    # - True: a skrl trainer will be used to evaluate the agent
+    # - False: the interaction with the environment will be performed manually.
+    #          This mode allows recording specific information about the environment
+
+    # configure and instantiate the RL trainer
+    # https://skrl.readthedocs.io/en/latest/modules/skrl.trainers.sequential.html
+    if experiment_cfg["trainer"]["use_api"]:
+        trainer_cfg = experiment_cfg["trainer"]
+        trainer_cfg["disable_progressbar"] = True
+        trainer = ManualTrainer(cfg=trainer_cfg, env=env, agents=agent)
+
+        # simulate environment
+        while simulation_app.is_running():
+            # agent and environment stepping
+            trainer.eval()
+            # check if simulator is stopped
+            if env.sim.is_stopped():
+                break
+
+    # execute the interaction with the environment without using a trainer
+    # (note that skrl requires the execution of some additional agent methods for proper operation).
+    # https://skrl.readthedocs.io/en/latest/modules/skrl.trainers.manual.html
+    else:
+        timestep = 0
+        agent.init()
+        agent.set_running_mode("eval")
+
+        # reset environment
+        obs, infos = env.reset()
+        # simulate environment
+        while simulation_app.is_running():
+            timestep += 1
+            # agent stepping
+            actions = agent.act(obs, timestep=timestep, timesteps=None)[0]
+            # env stepping
+            next_obs, rewards, terminated, truncated, infos = env.step(actions)
```

Comment:
After every env step call, it is important to check if the simulator is stopped or not and exit right away. The main reason is that, in Isaac Sim, if the user presses stop, then all the physics handles are disengaged and we can no longer read any new data from the physics buffers. Thus, inside the `IsaacEnv` class, we check for this and return the last values that were stored in the buffers. This logic is defined [here](https://github.com/NVIDIA-Omniverse/Orbit/blob/main/source/extensions/omni.isaac.orbit_envs/omni/isaac/orbit_envs/isaac_env.py#L246-L275).

Thus, as a practice, I would suggest something similar to [here](https://github.com/NVIDIA-Omniverse/Orbit/blob/main/source/standalone/workflows/rl_games/play.py#L128-L132):

```python
# env stepping
obs, _, dones, _ = env.step(actions)
# check if simulator is stopped
if env.unwrapped.sim.is_stopped():
    break
```

Reply:
This is called at the end of each iteration loop:

```python
            # check if simulator is stopped
            if env.sim.is_stopped():
                break
```

All methods that are called after `env.step` and before this, belong to agent and don't interact with the environment

Reply:
But I can move it just after the environment stepping

Reply:
Yeah didn't realize this was just how the trainer class works. Makes sense now for consistent behavior.